### PR TITLE
Add signature_algorithm_oid field to SignedData.

### PR DESCRIPTION
**Pull request by SanjayVas**

### CODE REVIEW

**Comment by wfa-reviewable** [comment]
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/177)

**Review by stevenwarejones — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/177)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @SanjayVas)

___
*[`src/main/proto/wfa/measurement/api/v2alpha/crypto.proto` line 51 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/177#-NfWI2Qp6cZs79Zyt2s8:-NfWI2Qp6cZs79Zyt2s9:b-ijdl3b) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/e2b2cf0f1b6867658dd446d31fe0b1bc16897f86/src/main/proto/wfa/measurement/api/v2alpha/crypto.proto#L51)):*
> ```proto
>   // If not specified, this is assumed to be the signature algorithm of the
>   // accompanying certificate.
>   string signature_algorithm_oid = 3;
> ```

a little harder to implement but it might be better to only specify the hash algorithm since the encryption algorithm type (rsa vs ec, etc) is already determined by the public key on the certificate.

**Review by stevenwarejones — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/177)* status: 0 of 1 files reviewed, all discussions resolved

**Review by SanjayVas — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/177)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @stevenwarejones)

___
*[`src/main/proto/wfa/measurement/api/v2alpha/crypto.proto` line 51 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/177#-NfWI2Qp6cZs79Zyt2s8:-NfXEAcy7yRG7qTrqwvQ:b-ft423h) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/e2b2cf0f1b6867658dd446d31fe0b1bc16897f86/src/main/proto/wfa/measurement/api/v2alpha/crypto.proto#L51)):*
<details><summary><i>Previously, stevenwarejones (Steven Ware Jones) wrote…</i></summary><blockquote>

a little harder to implement but it might be better to only specify the hash algorithm since the encryption algorithm type (rsa vs ec, etc) is already determined by the public key on the certificate.
</blockquote></details>

tl;dr there are actually multiple possible signature algorithms for the same key type and hash algorithm

This is not technically true, though we don't currently support this algorithm as it's a bit more complicated: RSA-PSS. This is a separate signature algorithm for RSA keys that can use the same hash algorithm. You *can* also technically use a key type that's specific to RSA-PSS, but that's not required and it works just fine with a regular RSA key.

**Review by stevenwarejones — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/177)* status: 0 of 1 files reviewed, all discussions resolved

___
*[`src/main/proto/wfa/measurement/api/v2alpha/crypto.proto` line 51 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/177#-NfWI2Qp6cZs79Zyt2s8:-NfXbr_k4mW5fEQHBzt6:b25duq4) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/e2b2cf0f1b6867658dd446d31fe0b1bc16897f86/src/main/proto/wfa/measurement/api/v2alpha/crypto.proto#L51)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

tl;dr there are actually multiple possible signature algorithms for the same key type and hash algorithm

This is not technically true, though we don't currently support this algorithm as it's a bit more complicated: RSA-PSS. This is a separate signature algorithm for RSA keys that can use the same hash algorithm. You *can* also technically use a key type that's specific to RSA-PSS, but that's not required and it works just fine with a regular RSA key.
</blockquote></details>

yeah - I agree. lets keep this as-is.

**Review by SanjayVas — COMMENTED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/177)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @SanjayVas)
